django/cantusdb_project/main_app/management/commands/add_cantus_index_merge_events.py

```diff
@@ -0,0 +1,94 @@
+from typing import Optional
+from datetime import datetime
+from django.core.management.base import BaseCommand
+from main_app.models import Chant
+from cantusindex import get_merged_cantus_ids
+
+
+class Command(BaseCommand):
+    help = (
+        "Fetch and apply merge events from the /json-merged-chants API on Cantus Index"
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "date",
+            nargs="?",
+            type=str,
+            help="Filter merges by date in the format YYYY-MM-DD",
+        )
+
+    def handle(self, *args, **kwargs):
+        date_filter: str = kwargs["date"]
+        if date_filter:
+            try:
+                date_filter = datetime.strptime(date_filter, "%Y-%m-%d")
+            except ValueError:
+                self.stdout.write(
+                    self.style.ERROR("Invalid date format. Please use YYYY-MM-DD.")
+                )
+                return
+
+        # Fetch the merge events from the Cantus Index API
+        merge_events: Optional[list] = get_merged_cantus_ids()
+
+        if not merge_events:
+            self.stdout.write(
+                self.style.ERROR("Failed to fetch data from Cantus Index.")
+            )
+            return
+
+        # Filter merge events by the provided date
+        if date_filter:
+            merge_events = self.filter_merge_events_by_date(merge_events, date_filter)
+
+        # Apply the new merge events on Cantus DB
+        for tx in merge_events:
+            self.apply_transaction(tx)
+
+    def filter_merge_events_by_date(self, merge_events: list, date_filter: str) -> list:
```

Comment:
I think the type of `date_filter` won't be `str` here cause you already converted it to a datetime object at line 25.

Reply:
Right, thanks!